app/controllers/gigs_controller.rb

```diff
@@ -0,0 +1,92 @@
+class GigsController < ApplicationController
+  before_action :set_gig, only: [:show, :update, :destroy]
+
+  # GET /gigs
+  def index
+    brand_name = params[:brand_name]
+    creator_id = params[:creator_id]
+    @gigs = Gig.all
+    @gigs = @gigs.where(brand_name: brand_name) if brand_name
+    @gigs = @gigs.where(creator_id: creator_id) if creator_id
+
+    render json: @gigs
+
+    rescue ActiveRecord::RecordNotFound => e
+      logger.info(e)
+      render json: { error: e }, status: :not_found
+  end
+
+  # GET /gigs/1
+  def show
+    param! :include, String, in: ["gig_payment", "creator"]
+
+    render json: @gig, include: params[:include]
+
+    rescue RailsParam::InvalidParameterError => e
+      logger.info(e)
+      render json: { error: e }, status: :bad_request
```

Comment:
I'm wondering if it's not better to move this InvalidParameterError exception catching into the ApplicationController

look at this-
https://stackoverflow.com/a/6773038/19554595